src/rgw/services/svc_tier_rados.h

```diff
@@ -129,12 +129,13 @@ class RGWSI_Tier_RADOS : public RGWServiceInstance
   }
 
   static inline bool raw_obj_to_obj(const rgw_bucket& bucket, const rgw_raw_obj& raw_obj, rgw_obj *obj) {
-    ssize_t pos = raw_obj.oid.find('_');
+    string oid = raw_obj.oid.substr(bucket.marker.length());
```

Comment:
Rather than create a new string, the `find` function allows for a second parameter that specifies where in the string to begin the search.

Reply:
ok